### DISCUSSION TITLE
Feature Request: Make `Compile active file` a first-class citizen

### DISCUSSION THREAD

**Discussion by dslijepcevic** [microsoft/vscode-cmake-tools#3844] [Ideas]
At the moment, `Compile active file` is not well integrated into CMake Tools; it feels disconnected, like a side, one-off thing. From the user's standpoint, there's really no reason for the compilation of a single file to be treated differently than building any other target via CMake.

To achieve this, I believe that the output of the compilation command should be parsed, and all warnings and errors displayed in the Problems tab, replacing the last build's output. This is how things work in Visual Studio and other IDEs, i.e., compilation with Ctrl+F7 is seamless. Perhaps you can add an option for this if you don't want this behavior to be the default.

**Comment by gcampbell-msft** [comment]
@dslijepcevic Thanks for the suggestion, we would agree that this would be great, however, there are lots of different intricacies involved. For now, we have multiple different issues open for this, and we are tracking those. Specifically, this one: #3709
